woocommerce/woocommerce-android
# Issue/2844 order detail refactor update order status

This PR takes care of the 10/13 step of the Order Detail refactoring #2844 .

#### Changes
- Adds the option to update order status.
- Adds the option to Issue Refund for an order.
- Adds the option to view the Refund detail screen for an order.
- Adds the option to send an sms to the billing phone, if available.
- Adds the option to call the billing phone, if available.
- Adds the option to send email to the billing email, if available.

#### Notes
- I have split this massive change into multiple small PRs in order to make reviews easier. 
- This PR is to be merged into a feature branch.
- **This PR is in draft till this [Step 9 PR](https://github.com/woocommerce/woocommerce-android/pull/2874) can be merged.**

#### Testing
- Click on an order from the Orders tab.
  - Click on the Edit icon beside the Order status.
  - Change the order status to some other status and notice the Undo snackbar is displayed.
  - Click on the Undo Snackbar and notice the status is reverted.
  - Change the status again.
  - Notice that the status is updated successfully once the SnackBar is dismissed.
- Click on an order from the Orders tab.
  - Click on the Issue Refund button.
  - Notice that once the products are refunded, the order detail is refreshed to reflect that change.
  - Notice that a new order note is added to the order after a successful refund.
- Click on an order from the Orders tab.
  - Click on the Refunds section and notice the refund detail screen is displayed.

#### Screenshots
##### Update order status
<img src="https://user-images.githubusercontent.com/22608780/93732308-5c723c00-fbee-11ea-9233-dfba2cd5d217.gif" width="300"/>. 

##### Open Refund detail
<img src="https://user-images.githubusercontent.com/22608780/93732344-76138380-fbee-11ea-9c5e-97f96322897c.gif" width="300"/>. 

##### Issue refund
<img src="https://user-images.githubusercontent.com/22608780/93846545-f9ea7000-fcc1-11ea-9c02-47627df072c9.gif" width="300"/>. 

Update release notes:
